pytorch_pfn_extras/training/extensions/util.py

```diff
@@ -15,73 +22,84 @@ class _CONSOLE_SCREEN_BUFFER_INFO(ctypes.Structure):
                     ('srWindow', ctypes.wintypes.SMALL_RECT),
                     ('dwMaximumWindowSize', _COORD)]
 
-    def set_console_cursor_position(x, y):
+    def set_console_cursor_position(x: int, y: int) -> None:
         """Set relative cursor position from current position to (x,y)"""
 
-        whnd = ctypes.windll.kernel32.GetStdHandle(_STD_OUTPUT_HANDLE)
+        whnd = windll.kernel32.GetStdHandle(_STD_OUTPUT_HANDLE)
         csbi = _CONSOLE_SCREEN_BUFFER_INFO()
-        ctypes.windll.kernel32.GetConsoleScreenBufferInfo(whnd,
-                                                          ctypes.byref(csbi))
+        windll.kernel32.GetConsoleScreenBufferInfo(whnd, ctypes.byref(csbi))
         cur_pos = csbi.dwCursorPosition
         pos = _COORD(cur_pos.X + x, cur_pos.Y + y)
 
         # Workaround the issue that pyreadline overwrites the argtype
-        setpos = ctypes.windll.kernel32.SetConsoleCursorPosition
+        setpos = windll.kernel32.SetConsoleCursorPosition
         argtypes = setpos.argtypes
         setpos.argtypes = None
         setpos(whnd, pos)
         setpos.argtypes = argtypes
 
-    def erase_console(x, y, mode=0):
+    def erase_console(x: int, y: int, mode: int = 0) -> None:
         """Erase screen.
 
         Mode=0: From (x,y) position down to the bottom of the screen.
         Mode=1: From (x,y) position down to the beginning of line.
         Mode=2: Hole screen
         """
 
-        whnd = ctypes.windll.kernel32.GetStdHandle(_STD_OUTPUT_HANDLE)
+        whnd = windll.kernel32.GetStdHandle(_STD_OUTPUT_HANDLE)
         csbi = _CONSOLE_SCREEN_BUFFER_INFO()
-        ctypes.windll.kernel32.GetConsoleScreenBufferInfo(whnd,
-                                                          ctypes.byref(csbi))
+        windll.kernel32.GetConsoleScreenBufferInfo(whnd, ctypes.byref(csbi))
         cur_pos = csbi.dwCursorPosition
         wr = ctypes.c_ulong()
         if mode == 0:
             num = csbi.srWindow.Right * (
                 csbi.srWindow.Bottom - cur_pos.Y) - cur_pos.X
-            ctypes.windll.kernel32.FillConsoleOutputCharacterA(
+            windll.kernel32.FillConsoleOutputCharacterA(
                 whnd, ord(' '), num, cur_pos, ctypes.byref(wr))
         elif mode == 1:
             num = cur_pos.X
-            ctypes.windll.kernel32.FillConsoleOutputCharacterA(
+            windll.kernel32.FillConsoleOutputCharacterA(
                 whnd, ord(' '), num, _COORD(0, cur_pos.Y), ctypes.byref(wr))
         elif mode == 2:
             os.system('cls')
 
 
+class _QueUnit(NamedTuple):
+    iteration: int
+    epoch_detail: float
+    time: float
+
+
 class ProgressBar:
 
-    def __init__(self, out=None):
+    def __init__(self, out: Optional[TextIO] = None) -> None:
         self._out = sys.stdout if out is None else out
-        self._recent_timing = collections.deque([], maxlen=100)
+        self._recent_timing: 'queue.Queue[_QueUnit]' = queue.Queue(maxsize=100)
```

Comment:
Question: Why is this implementation changed?

Reply:
It is just a cosmetic change, not for a typing issue.
I can drop bb58bb88f20fb35016e6c70e0cb40d098c5d322b if it is not suitable for this pull request.